### PR TITLE
Issue #52

### PR DESCRIPTION
**Pull request by ajurasz**
I have create generic Docker deployable/execution handler and now we are supporting following URI:
<code>docker:dockerfile/mongodb[:tag]</code>
because it is so generic I came to conclusion that we don't need MongoDB deployable/execution handler any more.

There is also one important change for deploying docker - this command waits for image to be pulled from repository. In this way user know when image is ready for use. With previous implementation I received information that image was deployed but it was not present on <code>docker images</code> list.

For testing I have created new image [1] which is micro container (~2.5Mb, base on busybox [2])  because mongodb was to heavy for our testing.

I'm not sure only about performing tests on zed-deployer. Are we doing this in right way as we require docker to be running localy?

[1] - https://registry.hub.docker.com/u/ajurasz/busybox/
[2] - https://registry.hub.docker.com/_/busybox/

### CODE REVIEW

File: deployer/src/main/java/zed/deployer/handlers/BaseDockerDeployableHandler.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        if (dockerUri.length == 3) {
+            inputStream = docker.pullImageCmd(dockerUri[1]).withTag(dockerUri[2]).exec();
+        } else {
+            inputStream = docker.pullImageCmd(dockerUri[1]).exec();
+        }
+
+        asString(inputStream);
+    }
+
+    protected String asString(InputStream response)  {
```
Comment (hekonsek) [inline deployer/src/main/java/zed/deployer/handlers/BaseDockerDeployableHandler.java]: This method can be replaced with commons-io `IOUtils.toString(InputStream)` method.

**Comment by hekonsek** [comment]
Regarding Mongo deployer - I still would like to keep it :) . In the future we will add automatic MongoDB port exposure to it, default volume mounting (for example to `/var/mongodb`) and stuff like that. My idea is to just type `deploy mongodb:docker` to install MongoDB with all the reasonable defaults settings. We could also add some extra sharding [1] or HA [2] options to the `mongodb:docker` URI, so it worth to keep that deployment handler :) .

At last but not least, mongodb handler can be easily recognized by Zed shell to provide some extra MongoDB tooling for that deployable (like finding slow queries or other stats).

Can you add MongoDB handler again and address my `IOUtils#toString` comment above? I will review the pull request again then.

[1] http://docs.mongodb.org/manual/sharding/
[2] http://docs.mongodb.org/manual/core/replica-set-high-availability/

**Comment by hekonsek** [comment]
Oh and good job on catching that non-blocking Docker pull invocation. I wasn't aware of this :) .

**Comment by hekonsek** [comment]
Regarding Docker test - yes, you have to start Docker server to perform integration tests. BTW check out this [1] method - it skip the given test if Docker server is not present.

[1] https://github.com/hekonsek/zed/blob/4b7f10b57fd2f020d1aaeccf9dae9c7f630a646b/deployer/src/test/java/zed/deployer/executor/DefaultProcessExecutorTest.java#L45

**Comment by ajurasz** [comment]
Well now when I read your comment about Mongo, I'm starting to see some benefits of leaving these handlers. I will make appropriate changes to put it back.

Will address your comments shortly.

**Comment by ajurasz** [comment]
I restored MongoDb handlers which extends BaseDocker handlers, added <code>IOUtils#toString</code> and also made some cleaning of zed-deployer tests (so that <code>assumeTrue</code> will skip docker test)

**Comment by hekonsek** [comment]
I like it. Thanks! :)

**Pull request merged**